### ISSUE TITLE
Error 'Existing schema attributes cannot be modified or deleted' in 'amplify push' following 'amplify update auth'

### ISSUE THREAD

**Issue by rayshi67** [aws-amplify/amplify-cli#13160]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Java

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line


```

</details>

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

I couldn't push with AWS amplify as the Auth version was too old that needs to be updated.
I did 'amplify update auth' and selected 'Apply default configuration without Social Provider (Federation)'.

However after i did 'amplify push' I got the following error:

Existing schema attributes cannot be modified or deleted. (Service: AWSCognitoIdentityProvider; Status Code: 400; Error Code: InvalidParameterException; Request ID: null; Proxy: null)

### Reproduction steps (if applicable)

amplify update Auth
amplify push

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line

Deployment failed.
Deploying root stack tamappdriverandroid [ =================----------------------- ] 3/7
	amplify-tamappdriverandroid-t… AWS::CloudFormation::Stack     UPDATE_ROLLBACK_COMPLETE       Wed Aug 23 2023 16:00:33…     
	functiontamdriverAmplifyAuthP… AWS::CloudFormation::Stack     UPDATE_COMPLETE                Wed Aug 23 2023 15:59:57…     
	storages3tamdrivers3firmwareb… AWS::CloudFormation::Stack     UPDATE_COMPLETE                Wed Aug 23 2023 15:59:47…     
	authtamdriverAmplifyAuth       AWS::CloudFormation::Stack     UPDATE_FAILED                  Wed Aug 23 2023 15:59:43…     
Deploying api amplifyDatasource [ ---------------------------------------- ] 0/26
Deploying auth userPoolGroups [ ---------------------------------------- ] 0/2
Deploying auth tamdriverAmplifyAuth [ ---------------------------------------- ] 0/7
	UserPoolClientRole             AWS::IAM::Role                 UPDATE_FAILED                  Wed Aug 23 2023 15:59:38…     
	UserPool                       AWS::Cognito::UserPool         UPDATE_FAILED                  Wed Aug 23 2023 15:59:38…     
Deployed function tamdriverAmplifyAuthPostConfirmation [ ======================================== ] 3/3
Deploying function tamdriverAmplifyTripReceiptEmail [ ---------------------------------------- ] 0/5
Deployed storage s3tamdrivers3firmwarebucket [ ======================================== ] 7/7

🛑 The following resources failed to deploy:
Resource Name: UserPool (AWS::Cognito::UserPool)
Event Type: update
Reason: Existing schema attributes cannot be modified or deleted. (Service: AWSCognitoIdentityProvider; Status Code: 400; Error Code: InvalidParameterException; Request ID: null; Proxy: null)
URL: https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-2#/stacks/arn%3Aaws%3Acloudformation%3Aap-southeast-2%3A334851156215%3Astack%2Famplify-tamappdriverandroid-test-234322-authtamdriverAmplifyAuth-15Q51FO8UWH94%2F0f302b80-7f02-11ec-9b16-02a200717f74/events


🛑 Resource is not in the state stackUpdateComplete
Name: UserPool (AWS::Cognito::UserPool), Event Type: update, Reason: Existing schema attributes cannot be modified or deleted. (Service: AWSCognitoIdentityProvider; Status Code: 400; Error Code: InvalidParameterException; Request ID: null; Proxy: null), IsCustomResource: false


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

### Additional information and screenshots

amplify --version
12.3.0

**Comment by ykethan** [comment]
Hey @ankpshah, thank you for reaching out. The issue appears to be similar to https://github.com/aws-amplify/amplify-cli/issues/13095 and https://github.com/aws-amplify/amplify-cli/issues/12827. 

Was the Amplify CLI upgraded from version before 7.x? If yes, could you try adding the attributes using the override functionality by running `amplify override auth`. similar to https://github.com/aws-amplify/amplify-cli/issues/13095 and https://github.com/aws-amplify/amplify-cli/issues/12827. 
Do let us know if this does not mitigate the issue.

**Comment by rayshi67** [comment]
Hi @ykethan 

Thank you for looking into the issue which is currently blocking the development work.

The Amplify CLI was upgraded from version 10.4.1. We are having the same error on Windows 10, as well as MacBook.

**Comment by rayshi67** [comment]
**Further update:**

I followed the suggestion in #12827, and added the custom user pool schema attribute to override.ts following the 'amplify override auth' command.

It now gives the following authorisation error,

`Deploying auth userPoolGroups [ ---------------------------------------- ] 0/2
    driverGroup                    AWS::Cognito::UserPoolGroup    UPDATE_FAILED                  Tue Aug 29 2023 14:58:44…     
Deployed auth tamdriverAmplifyAuth [ ======================================== ] 7/7
    UserPoolClientRole             AWS::IAM::Role                 UPDATE_IN_PROGRESS             Tue Aug 29 2023 14:59:10…     
    UserPool                       AWS::Cognito::UserPool         UPDATE_COMPLETE                Tue Aug 29 2023 14:59:08…     
    UserPoolClientWeb              AWS::Cognito::UserPoolClient   UPDATE_IN_PROGRESS             Tue Aug 29 2023 14:59:08…     
    UserPoolClient                 AWS::Cognito::UserPoolClient   UPDATE_COMPLETE                Tue Aug 29 2023 14:59:09…     
    IdentityPoolRoleMap            AWS::Cognito::IdentityPoolRol… UPDATE_COMPLETE                Tue Aug 29 2023 14:58:14…     
Deployed function tamdriverAmplifyAuthPostConfirmation [ ======================================== ] 3/3
Deploying function tamdriverAmplifyTripReceiptEmail [ ---------------------------------------- ] 0/5
Deployed storage s3tamdrivers3firmwarebucket [ ======================================== ] 7/7

🛑 The following resources failed to deploy:
Resource Name: driverGroup (AWS::Cognito::UserPoolGroup)
Event Type: update
Reason: User: arn:aws:sts::334851156215:assumed-role/ap-southeast-2_nvb1BLTw0_Full-access/amplifyadmin is not authorized to perform: cognito-idp:UpdateGroup on resource: arn:aws:cognito-idp:ap-southeast-2:334851156215:userpool/ap-southeast-2_BNlfgwIfG because no identity-based policy allows the cognito-idp:UpdateGroup action (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: AccessDeniedException; Request ID: 972bbe73-0836-4f3c-a3ff-72c553267a78; Proxy: null)
URL: https://console.aws.amazon.com/cloudformation/home?region=ap-southeast-2#/stacks/arn%3Aaws%3Acloudformation%3Aap-southeast-2%3A334851156215%3Astack%2Famplify-tamappdriverandroid-test-234322-authuserPoolGroups-SM517PZYJBRD%2F7f785f20-7f02-11ec-a143-0238336d1d20/events


🛑 Resource is not in the state stackUpdateComplete
Name: driverGroup (AWS::Cognito::UserPoolGroup), Event Type: update, Reason: User: arn:aws:sts::334851156215:assumed-role/ap-southeast-2_nvb1BLTw0_Full-access/amplifyadmin is not authorized to perform: cognito-idp:UpdateGroup on resource: arn:aws:cognito-idp:ap-southeast-2:334851156215:userpool/ap-southeast-2_BNlfgwIfG because no identity-based policy allows the cognito-idp:UpdateGroup action (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: AccessDeniedException; Request ID: 972bbe73-0836-4f3c-a3ff-72c553267a78; Proxy: null), IsCustomResource: false`

**Comment by rayshi67** [comment]
I deleted a couple of old environments yesterday before doing a clone, and I can see all deletes are completed successfully on Cloudformation console. I cannot see any relevant error there.

There was no error occurred using 'amplify add env' and 'amplify push' earlier.

I tried again to clone the existing env today, and got back the same error,
'Cloning env:test to env: tamdev in app:d24pb2d1td9ltj failed'

**Comment by ykethan** [comment]
Hey @rayshi67, wanted to get some additional information in using the clone mechanism in the AWS console? 
As using `amplify add env` and `amplify push` should create a new environment similar to your existing resources. 
Additionally, could you also provide us additional information on the migration scripts and its usage?

**Comment by rayshi67** [comment]
So what is the root cause of the error when doing the clone in Amplify console?

Without using the clone in Amplify console, how could we back up the Amplify environment including its data?

And if we use ‘amplify env add’ to spin a new  env, how would the data in cognito, dynamodb, and s3 etc in existing amplify env get populated into the newly created env? At the moment we haven’t got any mechanism in place for achieving that. So it would be great if you could point us to the right direction.

**Comment by ykethan** [comment]
@rayshi67 thank you for the information. The clone functionality in the AWS Amplify console is similar to `amplify add env`, it does not clone the data for the underlying resources being created. As you have pointed, we may have to utilzie custom scripts or AWS services that may assist in cloning the data.
additionally, the clone error appears to be similar to https://github.com/aws-amplify/amplify-hosting/issues/3022. I have reached out to our folks in hosting team and will provide an update as soon as I receive any additional information.

**Comment by ykethan** [comment]
wanted to provide an update on this issue, I was able to reproduce the issue by diving into the logs. The clone error was occurring due to resource overrides being present in the project. I have reached out to the appropriate Amplify team regarding this behaviour. 
But we should be able to utilize `amplify add env` to create a environment with similar resources as a workaround. Thank you for your patience.
